docs/advanced-node-scheduling.asciidoc

```diff
@@ -0,0 +1,342 @@
+[id="{p}-advanced-node-scheduling"]
+
+== Advanced Elasticsearch node scheduling
+
+Elastic Cloud on Kubernetes (ECK) offers full control over cluster nodes scheduling by combining Elasticsearch configuration with Kubernetes scheduling options:
+
+* defining Elasticsearch nodes roles
```

Comment:
We can link these to their relevant sections in the doc. Maybe in a separate PR?